### ISSUE TITLE
Messing up a shellcraft command resets the context archeitecture

### ISSUE THREAD

**Issue by toomanybananas** [binjitsu/binjitsu#72]
Hello,

While writing some bad shellcode today I noticed something weird. If you set context.arch to be amd64 and then screw up a shellcraft call, it will reset context.arch back to i386

To reproduce:

1. Start a new interactive python session, and enter from pwn import *
2. Set arch to amd64: context.arch = 'amd64'
3. Screw up a shellcraft command: shellcraft.open("/flag", 0, 0)
4. Check the value of context.arch, which should be i386

Tested on the latest git version of Binjitsu (as of time of writing) on Debian 8 x64

**Comment by zachriggle** [comment]
Hmm, I couldn't reproduce this.

```py
from pwn import *

context.arch='amd64'

print context

try:
    shellcraft.open("/flag", 0, 0)
except Exception as e:
    print e

print context
```

From that, I get:

```
ContextType(arch = 'amd64', bits = 64, endian = 'little')
/flag is not a valid expression
ContextType(arch = 'amd64', bits = 64, endian = 'little')
```

**Comment by zachriggle** [comment]
Oh, it does reproduce in the interactive terminal.  Interestingly, this reproduces in `python` but not `ipython`.

**Comment by zachriggle** [comment]
And it doesn't reproduce in `python` interactive terminal with my script.  It looks like there's something weird with not having an exception handler.

**Comment by zachriggle** [comment]
Fixed with d3ab883

**Comment by zachriggle** [comment]
Re-fixed with 3192839
